src/meta/src/controller/fragment.rs

```diff
@@ -1057,32 +1063,50 @@ impl CatalogController {
         } else {
             filter_condition
         };
-        let actor_info: Vec<(
-            ActorId,
-            WorkerId,
-            FragmentId,
-            I32Array,
-            DatabaseId,
-            ObjectId,
-        )> = Actor::find()
+        #[expect(clippy::type_complexity)]
+        let mut actor_info_stream: BoxStream<
+            '_,
+            Result<
+                (
+                    ActorId,
+                    WorkerId,
+                    FragmentId,
+                    StreamNode,
+                    I32Array,
+                    DatabaseId,
+                    ObjectId,
+                ),
+                _,
+            >,
+        > = Actor::find()
             .select_only()
             .column(actor::Column::ActorId)
             .column(actor::Column::WorkerId)
             .column(fragment::Column::FragmentId)
+            .column(fragment::Column::StreamNode)
             .column(fragment::Column::StateTableIds)
             .column(object::Column::DatabaseId)
             .column(object::Column::Oid)
             .join(JoinType::InnerJoin, actor::Relation::Fragment.def())
             .join(JoinType::InnerJoin, fragment::Relation::Object.def())
             .filter(filter_condition)
             .into_tuple()
-            .all(&inner.db)
+            .stream(&inner.db)
```

Comment:
According to the doc the streaming query will hold the connection until it's dropped. I'm not sure if there will be any performance issues, but using it to load all actors during the recovery phase should be okay.